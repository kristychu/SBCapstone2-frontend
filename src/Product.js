import React from "react";

function Product({ id, brand, name }) {
  return (
    <span key={id}>
      {brand} {name}
    </span>
  );
}

export default Product;
