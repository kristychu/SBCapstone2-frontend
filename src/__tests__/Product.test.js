import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers/rootReducer";
import { render } from "@testing-library/react";
import Product from "../Product";
import { MemoryRouter } from "react-router";

const store = createStore(reducer);

it("should render without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <Product />
      </MemoryRouter>
    </Provider>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Product />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
