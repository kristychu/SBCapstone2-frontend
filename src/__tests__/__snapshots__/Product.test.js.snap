// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      <b>
        Product Brand:
      </b>
       
    </p>
    <p>
      <b>
        Product Name:
      </b>
       
    </p>
  </div>
</DocumentFragment>
`;
